File: Easy/217_Contains_Duplicate/Solution.py
from typing import List, Optional


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()

        for num in nums:
            if num in seen:
                return True
            seen.add(num)

        return False
        


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    nums1 = [1, 2, 3, 1]
    print(solution.containsDuplicate(nums1))
    
    # test cases 2
    nums2 = [1, 2, 3, 4]
    print(solution.containsDuplicate(nums2))

    # test cases 3
    nums3 = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2]
    print(solution.containsDuplicate(nums3))
    
    

File: Easy/217_Contains_Duplicate/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int, int> mp;

        for (int i = 0; i < nums.size(); i++) {
            mp[nums[i]]++;

            if (mp[nums[i]] >= 2) {
                return true;
            }
        }
        return false;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    vector<int> nums1 = {1,2,3,1};
    cout << solution.containsDuplicate(nums1) << endl;

    // test cases 2
    vector<int> nums2 = {1,2,3,4};
    cout << solution.containsDuplicate(nums2) << endl;

    // test cases 3
    vector<int> nums3 = {1,1,1,3,3,4,3,2,4,2};
    cout << solution.containsDuplicate(nums3) << endl;

    return 0;
}